numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -0,0 +1,460 @@
+# SPDX-FileCopyrightText: 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+from inspect import signature
+from warnings import warn
+
+import dpctl
+import dpctl.program as dpctl_prog
+from numba.core.types import Array as ArrayType
+
+from numba_dpex import config
+from numba_dpex.core.descriptor import dpex_target
+from numba_dpex.core.exceptions import (
+    ComputeFollowsDataInferenceError,
+    ExecutionQueueInferenceError,
+    IllegalRangeValueError,
+    InvalidKernelLaunchArgsError,
+    SUAIProtocolError,
+    UnknownGlobalRangeError,
+    UnsupportedBackendError,
+    UnsupportedNumberOfRangeDimsError,
+    UnsupportedWorkItemSizeError,
+)
+from numba_dpex.core.kernel_interface.arg_pack_unpacker import Packer
+from numba_dpex.core.kernel_interface.spirv_kernel import SpirvKernel
+from numba_dpex.dpctl_iface import USMNdArrayType
+
+
+def get_ordered_arg_access_types(pyfunc, access_types):
```

Comment:
What if we make these kind of functions static? Under some class?

Reply:
We will remove this particular function as soon as we stop supporting NumPy array as kernel arguments. So, let it be for the time being.